tools/clang/unittests/HLSL/ExecutionTest.cpp

```diff
@@ -9740,99 +9749,141 @@ TEST_F(ExecutionTest, DynamicResourcesUniformIndexingTest) {
+  float expectedResultsUniform[expectedResultsSize] = {
+    10.0, 10.0, 
+    12.0, 12.0,
+    14.0, 14.0, 
+    20.0, 20.0, 
+    22.0, 22.0,
+    24.0, 24.0, 
+    30.0, 30.0, 
+    32.0, 32.0};
+
+  float expectedResultsNonUniform[expectedResultsSize] = {
+    10.0, 11.0, 
+    12.0, 13.0,
+    14.0, 15.0, 
+    20.0, 21.0, 
+    22.0, 23.0,
+    24.0, 25.0, 
+    30.0, 31.0, 
+    32.0, 33.0};
+  float *expectedResultsArray[2];
+  expectedResultsArray[0] = expectedResultsUniform;
+  expectedResultsArray[1] = expectedResultsNonUniform;
+    
+  for (unsigned int non_uniform_bit = 0; non_uniform_bit < 2; non_uniform_bit++) {
+    for (unsigned i = 0; i < _countof(TestShaderModels); i++) {
+      D3D_SHADER_MODEL sm = TestShaderModels[i];
+      LogCommentFmt(L"\r\nVerifying Dynamic Resources Uniform Indexing in shader "
+                    L"model 6.%1u",
+                    ((UINT)sm & 0x0f));
 
-  for (unsigned i = 0; i < _countof(TestShaderModels); i++) {
-    D3D_SHADER_MODEL sm = TestShaderModels[i];
-    LogCommentFmt(L"\r\nVerifying Dynamic Resources Uniform Indexing in shader "
-                  L"model 6.%1u",
-                  ((UINT)sm & 0x0f));
+      CComPtr<ID3D12Device> pDevice;
+      if (!CreateDevice(&pDevice, sm, false /* skipUnsupported */)) {
+        continue;
+      }
+      D3D12_FEATURE_DATA_D3D12_OPTIONS devOptions;
+      VERIFY_SUCCEEDED(
+          pDevice->CheckFeatureSupport((D3D12_FEATURE)D3D12_FEATURE_D3D12_OPTIONS,
+                                       &devOptions, sizeof(devOptions)));
+      if (devOptions.ResourceBindingTier < D3D12_RESOURCE_BINDING_TIER_3) {
+        WEX::Logging::Log::Comment(
+            L"Device does not support Resource Binding Tier 3");
+        WEX::Logging::Log::Result(WEX::Logging::TestResults::Skipped);
+        return;
+      }
 
-    CComPtr<ID3D12Device> pDevice;
-    if (!CreateDevice(&pDevice, sm, false /* skipUnsupported */)) {
-      continue;
-    }
-    D3D12_FEATURE_DATA_D3D12_OPTIONS devOptions;
-    VERIFY_SUCCEEDED(
-        pDevice->CheckFeatureSupport((D3D12_FEATURE)D3D12_FEATURE_D3D12_OPTIONS,
-                                     &devOptions, sizeof(devOptions)));
-    if (devOptions.ResourceBindingTier < D3D12_RESOURCE_BINDING_TIER_3) {
-      WEX::Logging::Log::Comment(
-          L"Device does not support Resource Binding Tier 3");
-      WEX::Logging::Log::Result(WEX::Logging::TestResults::Skipped);
-      return;
-    }
+      // Add compile options
 
-    // Test Compute shader
-    {
-      pShaderOp->CS = pShaderOp->GetString("CS66");
-      std::shared_ptr<ShaderOpTestResult> test = RunShaderOpTestAfterParse(
-          pDevice, m_support, "DynamicResourcesUniformIndexing", nullptr,
-          ShaderOpSet);
+      std::string compilerOptions = "";
+      if (sm==D3D_SHADER_MODEL_6_0)
+        compilerOptions += "-D FALLBACK=1";
+      if (non_uniform_bit && sm==D3D_SHADER_MODEL_6_0)
+        compilerOptions += ", -D NON_UNIFORM=1";
```

Comment:
You don't want a `,` in the string between options here, only a space.  If you simply add a leading space for the `" -D NON_UNIFORM=1"` string (as I had written in my comment), you don't need this branch at all.  Was there a problem running with the options from my previous comment?  Was the leading space a problem?  If so, just try trailing space instead.